### ISSUE TITLE
Update the Worker template to use the `Host.CreateApplicationBuilder` API.

### ISSUE THREAD

**Issue by IEvangelist** [dotnet/aspnetcore#43113]
Update the Worker template to use the `Host.CreateApplicationBuilder` API.

**Comment by eerhardt** [comment]
We should reconsider updating the Worker template to `Host.CreateApplicationBuilder` in .NET 8. The callback approach that `Host.CreateDefaultBuilder` uses is no longer recommended going forward, and the programming model is no longer used in ASP.NET apps - it uses the `WebApplication.CreateBuilder()` API which is imperative, top-to-bottom programming. The Worker template should follow the same pattern.

**Comment by eerhardt** [comment]
cc @davidfowl @halter73 @DamianEdwards

**Comment by davidfowl** [comment]
Agreed, can we discuss the IHostBuilder API missing 😄

**Comment by eerhardt** [comment]
I don’t want to support it on HostApplicationBuilder for the same reasons as last release. It puts the 2 programming models together that are just not compatible. We will have the same issues that we did in aspnet core in 6.0.

Serilog was the only place we found that needed to change, and we have an issue logged to add IServiceCollection overloads.
https://github.com/serilog/serilog-extensions-hosting/issues/66
